cmd/metal-api/internal/metal/provisioning.go

```diff
@@ -100,42 +134,40 @@ func (p provisioningEventSequence) containsEvent(event ProvisioningEventType) bo
 	return result
 }
 
-func (e *ProvisioningEvent) hasExpectedSuccessor(log *zap.SugaredLogger, actualSuccessor ProvisioningEventType) bool {
-	currentEvent := e.Event
-
-	expectedSuccessors, ok := expectedSuccessorEventMap[currentEvent]
-	if !ok {
-		log.Errorw("successor map does not contain an expected successor for event", "event", currentEvent)
-		return false
+func postEvent(currentFSM *fsm.FSM, event ProvisioningEventType) (*fsm.FSM, error) {
```

Comment:
You return `*fsm.FSM` but the caller does not use it. So I think there is no need to pass it around. I guess you want to make this function the place where you handle a new incoming machine event? Then maybe just pass the incoming event and use the signature `HandleMachineEvent(event ProvisioningEventType) error` or something? Later, we will probably also pass the machine entity, too, such that we can update the provisioning event container in the callback functions.